src/raglite/_database.py

```diff
@@ -331,53 +344,69 @@ def create_database_engine(config: RAGLiteConfig | None = None) -> Engine:
         with Session(engine) as session:
             metrics = {"cosine": "cosine", "dot": "ip", "euclidean": "l2", "l1": "l1", "l2": "l2"}
             session.execute(
-                text("""
-                CREATE INDEX IF NOT EXISTS keyword_search_chunk_index ON chunk USING GIN (to_tsvector('simple', body));
-                """)
+                text(
+                    """
+                    CREATE INDEX IF NOT EXISTS keyword_search_chunk_index ON chunk USING GIN (to_tsvector('simple', body));
+                    """
+                )
             )
-            session.execute(
-                text(f"""
+            base_sql = f"""
                 CREATE INDEX IF NOT EXISTS vector_search_chunk_index ON chunk_embedding
                 USING hnsw (
                     (embedding::halfvec({embedding_dim}))
                     halfvec_{metrics[config.vector_search_index_metric]}_ops
                 );
                 SET hnsw.ef_search = {20 * 4 * 8};
-                SET hnsw.iterative_scan = {'relaxed_order' if config.reranker else 'strict_order'};
-                """)
-            )
+            """
+            # Add iterative scan if version >= 0.8.0
+            pgvector_version = _get_pgvector_version(session)
+            if pgvector_version and version.parse(pgvector_version) >= version.parse("0.8.0"):
+                sql = f"""{base_sql};
+                    SET hnsw.iterative_scan = {'relaxed_order' if config.reranker else 'strict_order'};
+                    """
+            else:
+                sql = f"{base_sql};"
+            session.execute(text(sql))
             session.commit()
     elif db_backend == "sqlite":
         # Create a virtual table for keyword search on the chunk table.
         # We use the chunk table as an external content table [1] to avoid duplicating the data.
         # [1] https://www.sqlite.org/fts5.html#external_content_tables
         with Session(engine) as session:
             session.execute(
-                text("""
+                text(
+                    """
                 CREATE VIRTUAL TABLE IF NOT EXISTS keyword_search_chunk_index USING fts5(body, content='chunk', content_rowid='rowid');
-                """)
+                """
+                )
```

Comment:
There are a number of these superfluous whitespace changes – are you running an older or newer version of ruff perhaps? I'd prefer to avoid nonfunctional changes in the PR if possible.

Reply:
Yes, that's the case, I updated ruff this morning for that reason.